Cargo.toml

```diff
@@ -51,16 +51,24 @@ borsh = { version = "0.10", default-features = false, optional = true }
 version          = "0.34"
 default-features = false
 
+[dependencies.cosmos-sdk-proto]
+version = "0.20.0"
+default-features = false
+
 [dev-dependencies]
 serde_json = "1.0.107"
 
 [features]
 default            = ["std", "client"]
 std                = ["prost/std", "bytes/std", "subtle-encoding/std", "base64/std", "flex-error/std", "ics23/std", "informalsystems-pbjson/std"]
-serde              = ["dep:serde", "ics23/serde", "informalsystems-pbjson"]
-client             = ["std", "dep:tonic", "tonic/codegen", "tonic/transport", "tonic/prost"]
+serde              = ["dep:serde", "dep:informalsystems-pbjson", "ics23/serde", "cosmos-sdk-proto/serde"]
```

Comment:
`cosmos-sdk-proto` doesn't currently isn't a `serde` feature, which seems like a problem for this.

Happy to do whatever `ibc-proto-rs` was doing before (`pbjson`?)

Reply:
Yeah this depends on my fork of cosmos-rust with pbjson integration: https://github.com/cosmos/cosmos-rust/compare/main...informalsystems:cosmos-rust:romac/extract-ibc-proto

Reply:
These two PRs together also depend on some tweaks in tendermint-proto and ics23.

Reply:
But yeah perhaps the facture would be better named as `pbjson`.

Reply:
This was only meant as a first sketch to be discussed with you and see if this reasonable at all or not, and see how to proceed from here. So I am happy to hear your thoughts and any concerns you may have.

Reply:
I also wonder if we should move away from `pbjson-build` in cosmos-rust and use this [`protoc-gen-prost-serde `](https://github.com/neoeinstein/protoc-gen-prost/blob/main/protoc-gen-prost-serde/README.md ) buf plugin instead?

Perhaps best discussed once we have a PR on `cosmos-rust`.

Reply:
Either approach (`pbjson` or `protoc-gen-prost-serde`) seems reasonable to me, I was just confused how it was activating a `serde` feature that doesn't exist upstream. Makes sense now.

We can discuss on a `cosmos-sdk-proto` PR whenever you're ready to open that.

Reply:
Sounds good, will do that early next week week and we can discuss there :)